classes/log/store.php

```diff
@@ -63,15 +64,59 @@ protected function is_event_ignored(event_base $event) {
         return $isdisabledevent;
     }
 
+    /**
+     * get eventid from logstore_standard_log
+     * @param event_base $event
+     * @return int
+     *
+     */
+    protected function get_event_id($event) {
+        global $DB;
+
+        $sqlparams = [];
+        $sqlparams['eventname'] = $event->eventname;
+        $sqlparams['component'] = $event->component;
+        $sqlparams['action'] = $event->action;
+        $sqlparams['target'] = $event->target;
+        $sqlparams['objecttable'] = $event->objecttable;
+        $sqlparams['objectid'] = $event->objectid;
+        $sqlparams['timecreated'] = $event->timecreated;
+        $sqlparams['userid'] = $event->userid;
+        $sqlparams['anonymous'] = $event->anonymous;
+
+        $rows = $DB->get_records_sql("SELECT MAX(ID) AS id
+                    FROM {logstore_standard_log}
+                    WHERE eventname = :eventname
+                    AND component = :component
+                    AND action = :action
+                    AND target = :target
+                    AND (objecttable = :objecttable OR objecttable IS NULL)
+                    AND (objectid = :objectid OR objectid IS NULL)
+                    AND timecreated = :timecreated
+                    AND userid = :userid
+                    AND anonymous = :anonymous", $sqlparams);
+
+        if (empty($rows)) {
+            return 0;
+        }
+
+        $rowid = 0;
+        foreach ($rows as $row) {
```

Comment:
is there any need to loop here?

Reply:
The returned row(s) contains an associative array keyd be the id of the first returned row. There might be no rows, so the quickest way is to iterate and return the first row.

Reply:
since you know `$rows` is not empty, can't you just do `$rows[0]->id`?

Reply:
In Moodle get_records_xxxx  always return arrays that have the value of the primary key id as indexes so $row[0]->id would not work. We get stuff like this:
```
array(1) {
  [1487]=>
  object(stdClass)#221 (1) {
    ["id"]=>
    string(4) "1487"
  }
}

```
The easiest way to iterate this is with a for loop.

Reply:
Fair enough :)

Reply:
array_keys() and after that keys[0] but your solution is the same and fine for me.